environment/toolchain/v5/build.sh

```diff
@@ -0,0 +1,1234 @@
+
+# zsh does not recognize the option -t of the command type
+# therefore we use the alternative whence -w
+if [[ "\$ZSH_NAME" == "zsh" ]]; then
+    # check for active virtual environments
+    if [ "\$( whence -w deactivate )" != "deactivate: none" ]; then
+        echo \$env_error
+        return 0;
+    fi
+# any other shell
+else
+    # check for active virtual environments
+    if [ "\$( type -t deactivate )" != "" ]; then
+        echo \$env_error
+        return 0
+    fi
+fi
+
+# check that we aren't root
+if [[ "\$USER" == "root" ]]; then
+    echo "You shouldn't use the toolchain as root!"
+    return 0
+fi
+
+# save original environment
+export ORIG_PATH=\$PATH
+export ORIG_PS1=\$PS1
+export ORIG_LD_LIBRARY_PATH=\$LD_LIBRARY_PATH
+export ORIG_CXXFLAGS=\$CXXFLAGS
+export ORIG_CFLAGS=\$CFLAGS
+
+# activate new environment
+export PATH=$PREFIX:$PREFIX/bin:\$PATH
+export PS1="($NAME) \$PS1"
+export LD_LIBRARY_PATH=$PREFIX/lib:$PREFIX/lib64
+export CXXFLAGS=-isystem\ $PREFIX/include\ \$CXXFLAGS
+export CFLAGS=-isystem\ $PREFIX/include\ \$CFLAGS
+
+# disable root
+function su () {
+    echo "You don't want to use root functions while using the toolchain!"
+    return 1
+}
+function sudo () {
+    echo "You don't want to use root functions while using the toolchain!"
+    return 1
+}
+
+# create deactivation function
+function deactivate() {
+    export PATH=\$ORIG_PATH
+    export PS1=\$ORIG_PS1
+    export LD_LIBRARY_PATH=\$ORIG_LD_LIBRARY_PATH
+    export CXXFLAGS=\$ORIG_CXXFLAGS
+    export CFLAGS=\$ORIG_CFLAGS
+    unset ORIG_PATH ORIG_PS1 ORIG_LD_LIBRARY_PATH ORIG_CXXFLAGS ORIG_CFLAGS
+    unset -f su sudo deactivate
+}
+EOF
+fi
+
+BOOST_SHA256=4b2136f98bdd1f5857f1c3dea9ac2018effe65286cf251534b6ae20cc45e1847
+BOOST_VERSION=1.80.0
+BOOST_VERSION_UNDERSCORES=`echo "${BOOST_VERSION//./_}"`
+BZIP2_SHA256=a2848f34fcd5d6cf47def00461fcb528a0484d8edef8208d6d2e2909dc61d9cd
+BZIP2_VERSION=1.0.6
+DOUBLE_CONVERSION_SHA256=8a79e87d02ce1333c9d6c5e47f452596442a343d8c3e9b234e8a62fce1b1d49c
+DOUBLE_CONVERSION_VERSION=3.1.6
+FBLIBS_VERSION=2022.10.17.00
+FIZZ_SHA256=6c7069cb6812e9ed990b65e60c4d87b59d59c4a11f26d0ae1e35498e47489d9d
+FLEX_VERSION=2.6.4
+FMT_SHA256=5dea48d1fcddc3ec571ce2058e13910a0d4a6bab4cc09a809d8b1dd1c88ae6f2
+FMT_VERSION=9.1.0
+FOLLY_SHA256=651ba3ed2b38b02c604cf99e008c9e51d87e74c9af2da3c7eaee1240f72ac25b
+GFLAGS_COMMIT_HASH=b37ceb03a0e56c9f15ce80409438a555f8a67b7c
+GLOG_SHA256=eede71f28371bf39aa69b45de23b329d37214016e2055269b3b5e7cfd40b59f5
+GLOG_VERSION=0.5.0
+JEMALLOC_VERSION=5.3.0
+LIBAIO_VERSION=0.3.112
+LIBEVENT_VERSION=2.1.12-stable
+LIBSODIUM_VERSION=1.0.18
+LIBUNWIND_VERSION=1.6.2
+LZ4_SHA256=33af5936ac06536805f9745e0b6d61da606a1f8b4cc5c04dd3cbaca3b9b4fc43
+LZ4_VERSION=1.8.3
+PROXYGEN_SHA256=2f89e7d57d266504a191a74dad5f611c72467ab1bab077e0298368d7429901cb
+SNAPPY_SHA256=75c1fbb3d618dd3a0483bff0e26d0a92b495bbe5059c8b4f1c962b478b6e06e7
+SNAPPY_VERSION=1.1.9
+XZ_VERSION=5.2.5 # for LZMA
+ZLIB_VERSION=1.2.13
+ZSTD_VERSION=1.5.0
```

Comment:
There is also zstd 1.5.2, not sure how relevant that is for us

Reply:
I haven't updated that will do so 😄